integration_test/testgenerated/server/server-service.yml

```diff
@@ -34,6 +34,11 @@ services:
         args:
           body: optional<CustomObject>
         returns: optional<CustomObject>
+      echoOptionalAlias:
```

Comment:
Can you add a test case for an option alias that contains a collection? I believe this will introduce a compile time break since we check `returnVal == nil` in  `servicewriter.go` but the returnVal will be the optional type i.e. a struct which can't check for nil. So we may need to introspect on the `.Value` field there as well

Reply:
You are right! I've added a fix and test. If something is optional we don't want to check for nil at all (204s skip unmarshaling because there is no body) so I think it's most correct to remove it (rather than inspect the internal field at all)